### ISSUE TITLE
Administer users for all channels and groups from admin area

### ISSUE THREAD

**Issue by kenton-r** [RocketChat/Rocket.Chat#9611]
### Description:

Feature request for administration ability. There is a need for the administration to be able to add users to both public channels and private groups. This can be through the admin section.

### Expected behavior:

Administration actually be able to manage all activity.
The admin does not have to join the group, but must be able to manage the users of the group.
Could there be a user section, just like on the channels, in the admin -> rooms -> edit room section.

### Actual behavior:

There is no way for the administration to add users to or remove users from a private groups.

**Comment by richgansa** [comment]
The ability to add/remove users from private groups as an admin while not being a member would be extremely helpful. Our current solution is for the admin to be the owner with all notifications disabled and the room hidden.

**Comment by ChrissW-R1** [comment]
Yeah, this feature is really needed.
Currently we test two different solutions, which are both not the best, by different administrators.

The first solution is the one @richgansa talked about. This is ok, but the admin will be slowed down in its non-administrative work.

The second one is to leave the group. When a new user should added to this group the admin set the channel to public, joins it, add the user, leave the room and make it private again. But this is a very huge security breach. For a short time every user could be join the channel and get non-disclosure information!

**Comment by aba-rechsteiner** [comment]
+1

**Comment by aba-rechsteiner** [comment]
@rodrigok please have a look at this

**Comment by AmShaegar13** [comment]
May I add changing user roles here too? Regularly, the last owner of a channels leaves the company. I need to pass ownership by request. I could not find a way to do so without setting it in the database directly.

**Comment by piotrkochan** [comment]
Sure you can manage roles and add role to the user. Just edit user.

pon., 14 sty 2019, 18:02 użytkownik Finn Glöe <notifications@github.com>
napisał:

**Comment by ChrissW-R1** [comment]
Yes, it is possible to manage the roles via the administration GUI. But there is no option to add or remove users to/from channels! It's the same, to change the channel specific roles.

**Comment by FrancoisBillant** [comment]
I would like this too.

**Comment by devadattas** [comment]
Currently, there is a workaround to this issue:

1. Change the Private Group into a Channel
2. Join the channel and set self as "owner"
3. Edit it back to Private Group.

Hope this helps people.

**Comment by ChrissW-R1** [comment]
This is not a real solution. If the group is changed to an channel, every user could enter it. This is a very high risk!

**Comment by devadattas** [comment]
No doubt that this is a (very) risky workaround on busy environments, but this is a workaround none the less. 

It's better than modifying the db directly.

**Comment by jsternadel** [comment]
+1 for being able to add users to private groups as an admin.  The SlackBridge imports private rooms as private rooms without any local admin having access.  Had to follow @devadattas workaround to invite my users

**Comment by photoninger** [comment]
Indeed it is very annoying when you try do delete a user which is the last owner of a group or channel.

I would prefer that rocket.chat assigns automatically a new owner in this case:
If the group has at least one moderator, take on of them, 
if not, try the leaders and at last take one of the remaining members.
If the user is the last member, give admin the choice to delete the group.

**Comment by Bengrunt** [comment]
Having this feature would indeed be great !

**Comment by q16marvin** [comment]
we need it too!

**Comment by dcorrigangoempyrean** [comment]
I just encountered an issue where I need to set an announcement per our company policy to all channels and private groups.
Unfortunately, even though I am an admin, and can list all channels and private groups, I can't do anything with the private group without being a member. In my opinion, this is crippling the Admin role. I vote +1 for this issue!

**Comment by adolfombarros** [comment]
we look forward to this feature

**Comment by eskp** [comment]
The current workaround is to make the private channel public, admin joins the channel, makes the channel private, then she able to change any user permissions in that channel and leave the channel when done. As said previously in the busy env this can be a security risk but at least we can see in the channel who joins it.

**Comment by eengstrom** [comment]
The workaround is terrible.  That implies that I, as admin, have to be a member of the channel which means that I have to manage the notifications from those channels even if I never have any interest in it.  

"Admin" should be omnipotent, not crippled by a bad design.

**Comment by eskp** [comment]
@eengstrom You'd remove yourself from the channel straight after finishing your admin tasks so you never have to manage any notifications. This is just that, a workaround to get a task done until this is fixed in the admin UI.

**Comment by desser-code** [comment]
@sampaiodiego I think it would be really great if the administrator were really an administrator and could do such important things

**Comment by p1nkm4n** [comment]
nothing happend to this in the last 3 years? There is no way a Admin could manage users in privat channels? Very bad.....

**Comment by marek-ciazynski** [comment]
+1

**Comment by talss89** [comment]
This is, IMO, a key feature. Could we have an indication of where this sits in the development roadmap?

**Comment by docholoday** [comment]
Any movement on this? I'm trying to admin a group of middle-schoolers and shuffle users back and forth between rooms, without giving them the ability to make their own. Basic admin controls over rooms is pretty essential. We either need to be able to edit to room's users, or edit the users and add/remove them from rooms, whichever is easier.

**Comment by pierreozoux** [comment]
Relates/duplicate of https://github.com/RocketChat/Rocket.Chat/issues/17172

**Comment by drilichili** [comment]
is there any update on this?

**Comment by sandrotanner** [comment]
It is so annoying that they always tag issues as "triaged" but they never do this publicly but instead probably in some external tool so we are left guessing when (if at all) the issue will be tackled. Not really the nice FLOSS way of doing things if you ask me...
If it is out of scope just say it and move on so we know what to expect.
(end of rant :smile: )

**Comment by talss89** [comment]
@dougfabris Would it be possible to have some info on when this will be implemented, or even if it will? It does seem like pretty important functionality for a lot of people, us included.

Just trying to figure out if it's worth building a standalone tool to do this via the API.
